python/lsst/ts/watcher/mock_pager_duty.py

```diff
@@ -0,0 +1,239 @@
+            # ... use the server
+
+        # Or if you prefer explicit start/stop:
+
+        server = MockPagerDuty(port=...)
+        await server.start()
+        # ... use the server
+        await server.close()
+
+    Attributes
+    ----------
+    port : `int`
+        If the specified port is 0, then the chosen port
+        (after `start` has run). Otherwise the specified port.
+        The default is 80 because that is what PagerDuty uses.
+    url : `str`
+        The root URL of the service. "" until `start` has run.
+    """
+
+    ALLOWED_KEYS = frozenset(
+        (
+            "payload",
+            "routing_key",
+            "event_action",
+            "dedup_key",
+            "client",
+            "client_url",
+            "links",
+            "images",
+        )
+    )
+
+    ALLOWED_PAYLOAD_KEYS = frozenset(
+        (
+            "summary",
+            "timestamp",
+            "severity",
+            "source",
+            "component",
+            "group",
+            "class",
+            "custom_details",
+        )
+    )
+
+    def __init__(self, port: int = 80) -> None:
+        self.port = port
+        self.escalation_key = os.environ["ESCALATION_KEY"]
+        # Dict of event ID: data_dict
+        self.events: dict[str, dict] = dict()
+        self.runner: web.AppRunner | None = None
+        self.site: web.TCPSite | None = None
+        self.url: str = ""
+        # Unit tests may set this True to reject the next request
+        self.reject_next_request = False
+
+    def make_app(self) -> web.Application:
+        """Make an instance of the web application."""
+        app = web.Application()
+        app.add_routes(
+            [
+                # web.post("/:{identifier}/close", self.close_event),
+                web.post("/v2/enqueue", self.send_event),
+                # web.delete("/:{identifier}", self.delete_event),
+            ]
+        )
+        return app
+
+    async def start(self) -> None:
+        """Start the service.
+
+        Raises
+        ------
+        RuntimeError
+            If port = 0 and serving on more than one socket
+            (in which case the served port cannot be determined).
+        """
+        if self.runner is not None:
+            raise RuntimeError("Already started")
+        app = self.make_app()
+        self.runner = web.AppRunner(app)
+        await self.runner.setup()
+        self.site = web.TCPSite(self.runner, "127.0.0.1", port=self.port)
+        await self.site.start()
+        server = self.site._server
+        if self.port == 0:
+            if len(server.sockets) != 1:
+                raise RuntimeError(
+                    "Serving on more than one socket; cannot determine the port"
+                )
+            self.port = server.sockets[0].getsockname()[1]  # type: ignore
+        self.url = f"http://127.0.0.1:{self.port}"
+
+    async def close(self) -> None:
+        """Stop the service."""
```

Comment:
Please add a Raises section for the RuntimeError.

Reply:
Good point. I decide to omit the raise and silently work if already closed (for all three mock services).